src/client/websocket/handlers/APPLICATION_COMMAND_CREATE.js

```diff
@@ -3,18 +3,13 @@
 const { Events } = require('../../../util/Constants');
 
 module.exports = (client, { d: data }) => {
-  let command;
+  const commandManager = data.guild_id ? client.guilds.cache.get(data.guild_id)?.commands : client.application.commands;
+  if (!commandManager) return;
 
-  if (data.guild_id) {
-    const guild = client.guilds.cache.get(data.guild_id);
-    if (!guild) return;
-    command = guild.commands._add(data);
-  } else {
-    command = client.application.commands._add(data);
-  }
+  const command = commandManager._add(data, data.application_id === client.application.id);
 
   /**
-   * Emitted when an application command is created.
+   * Emitted when a guild application command is created.
```

Comment:
Not according to my testing, however I left the code for global commands in there in case they ever do get sent.. but the code can be removed of course.